File: Leetcode/14 - Longest Common Prefix/solution.py
def get_prefix(a: str, b: str) -> str:
    length = min(len(a), len(b))
    i = 0
    while i < length and a[i] == b[i]:
        i += 1
    return a[:i]

def longestCommonPrefix(strs: List[str]) -> str:
    prefix = strs[0]
    for i in range(1, len(strs)):
        prefix = get_prefix(prefix, strs[i])
        if not prefix:
            break
    return prefix

File: Leetcode/14 - Longest Common Prefix/solution.ts
function getPrefix(a: string, b: string): string {
    let len = Math.min(a.length, b.length);
    let i = 0;
    while (i < len && a[i] === b[i]) {
        i++;
    }
    return a.substring(0, i);
}

function longestCommonPrefix(strs: string[]): string {
    let prefix = strs[0];
    for (let i = 1; prefix.length > 0 && i < strs.length; i++) {
        prefix = getPrefix(prefix, strs[i]);
    }
    return prefix;
};